models/issue_comment.go

```diff
@@ -1164,3 +1168,58 @@ func getCommitIDsFromRepo(repo *Repository, oldCommitID, newCommitID, baseBranch
 
 	return
 }
+
+// CreateScheduledPRToAutoMergeComment creates a comment when a pr was set to auto merge once all checks succeed
+func CreateScheduledPRToAutoMergeComment(user *User, pr *PullRequest) (comment *Comment, err error) {
+
+	if pr.Issue == nil {
+		err = pr.LoadIssue()
+		if err != nil {
+			return
+		}
+	}
+
+	if pr.BaseRepo == nil {
+		err = pr.LoadBaseRepo()
+		if err != nil {
+			return
+		}
+	}
+
+	opts := &CreateCommentOptions{
+		Type:  CommentTypePRScheduledToAutoMerge,
+		Doer:  user,
+		Repo:  pr.BaseRepo,
+		Issue: pr.Issue,
+	}
+
+	comment, err = CreateComment(opts)
+	return
+}
+
+// CreateScheduledPRToAutoMergeComment creates a comment when a pr was set to auto merge once all checks succeed
+func CreateUnScheduledPRToAutoMergeComment(user *User, pr *PullRequest) (comment *Comment, err error) {
```

Comment:
I think this function is too similar to the previous one, maybe using one is enough.

Reply:
Moved it into a more general function.